nuxeo-features/nuxeo-automation/nuxeo-automation-features/src/test/java/org/nuxeo/ecm/automation/core/operations/services/bulk/TestSetPropertiesActionOptimizations.java

```diff
@@ -0,0 +1,188 @@
+
+    @Inject
+    public AuditLogger auditLogger;
+
+    protected DocumentModel note;
+
+    private class TestListener extends EventListenerDescriptor {
+
+        private int count;
+
+        private TestListener() {
+            events = new HashSet<>(auditLogger.getAuditableEventNames());
+        }
+
+        @Override
+        public void initListener() {
+            inLineListener = event -> count++;
+        }
+    }
+
+    @Before
+    public void before() {
+        DocumentModel workspace = session.getDocument(new PathRef("/default-domain/workspaces/test"));
+        DocumentModel note = session.createDocumentModel(workspace.getPathAsString(), "note", "Note");
+        this.note = session.createDocument(note);
+        transactions.nextTransaction();
+    }
+
+    @Test
+    public void testDisableAudit() throws InterruptedException {
+        TestListener listener = new TestListener();
+        eventService.addEventListener(listener);
+        assertEquals(0, listener.count);
+
+        // should increment listener count by 3
+        // documentCheckedIn, documentCreated and documentModified are fired
+        bulkService.submit(createBuilder().build());
+        assertTrue("Bulk action didn't finish", bulkService.await(Duration.ofSeconds(60)));
+        assertEquals(3, listener.count);
+
+        // should increment listener count by 1, as documentModified is not avoidable
+        // only documentModified is fired
+        bulkService.submit(createBuilder().param(PARAM_DISABLE_AUDIT, TRUE).build());
+        assertTrue("Bulk action didn't finish", bulkService.await(Duration.ofSeconds(60)));
+        assertEquals(4, listener.count);
+
+        // disabling versioning also reduce fired event number to 1, as documentModified is not avoidable
+        bulkService.submit(createBuilder().param(PARAM_VERSIONING_OPTION, NONE.toString()).build());
+        assertTrue("Bulk action didn't finish", bulkService.await(Duration.ofSeconds(60)));
+        assertEquals(5, listener.count);
+
+        eventService.removeEventListener(listener);
+    }
+
+    @Test
+    public void testDisableVersioning() throws InterruptedException {
+        assertEquals(1, session.getVersions(note.getRef()).size());
+        // should not create a new version
+        bulkService.submit(createBuilder().param(PARAM_VERSIONING_OPTION, NONE.toString()).build());
+        assertTrue("Bulk action didn't finish", bulkService.await(Duration.ofSeconds(60)));
+        transactions.nextTransaction();
+        assertEquals(1, session.getVersions(note.getRef()).size());
+        // should create a new version
+        bulkService.submit(createBuilder().build());
+        assertTrue("Bulk action didn't finish", bulkService.await(Duration.ofSeconds(60)));
+        transactions.nextTransaction();
+        assertEquals(2, session.getVersions(note.getRef()).size());
+    }
+
+    @Test
+    public void testParamValues() throws InterruptedException {
```

Comment:
I would be in favor to move this test close to `SetPropertiesAction` as it's about testing `remove*` method right?

Reply:
you should make the intent of the testing clear because it is not obvious that valid or invalid param result in the same assertion

Reply:
There the test was to prove that no value can break the computation, a problem we had recently on other computations.
What should I change to make this clear ?

Reply:
if you cannot express intent with code use comments